lmdeploy/pytorch/models/deepseek_v2.py

```diff
@@ -548,6 +825,126 @@ def forward(
         outputs = (hidden_states, residual)
         return outputs
 
+    def forward_yield(
+        self,
+        hidden_states: torch.Tensor,
+        rotary_pos_emb: Tuple[torch.FloatTensor, torch.FloatTensor],
+        past_key_value: Optional[List[torch.FloatTensor]],
+        residual: Optional[torch.Tensor] = None,
+        attn_metadata: Any = None,
+        tag: Any = None,
+    ):
+        """forward_yield."""
+        if residual is None:
+            residual = hidden_states
+            hidden_states = self.input_layernorm(hidden_states)
+        else:
+            hidden_states, residual = self.input_layernorm(hidden_states, residual)
+
+        # yield for attn0 and attn1
+        yield
+        # Self Attention
+        hidden_states = self.self_attn(
+            hidden_states=hidden_states,
+            rotary_pos_emb=rotary_pos_emb,
+            past_key_value=past_key_value,
+            attn_metadata=attn_metadata,
+        )
+
+        hidden_states, residual = self.post_attention_layernorm(hidden_states, residual)
+        batch_size, sequence_length, hidden_dim = hidden_states.shape
+        hidden_states = hidden_states.view(-1, hidden_dim)
+        topk_weights, topk_ids = self.mlp.gate(hidden_states)
+
+        exports = self.mlp.experts
+        topk_weights = exports.renormalize(topk_weights)
+        topk_weights = topk_weights.to(torch.float32)
+        topk_ids = topk_ids.to(torch.int64)
+        up_weight, up_scale, down_weight, down_scale = exports.get_weight()
```

Comment:
Some backends might change the layout of weights. We better not explicit get weights here.